market_maker_keeper/uniswapv2_market_maker_keeper.py

```diff
@@ -314,40 +330,50 @@ def add_liquidity(self) -> Optional[Receipt]:
                                     f"with tx hash {transact.transaction_hash.hex()}")
 
                 if self.uniswap.is_new_pool:
-                    self.uniswap.set_and_approve_pair_token(self.uniswap.get_pair_address(self.token_a.address, self.token_b.address))
+                    self.uniswap.set_pair_token(self.uniswap.get_pair_address(self.token_a.address, self.token_b.address))
 
-                # TODO: add calculation of our percentage of the pool
-                return transact
+                liquidity_tokens = self.uniswap.get_current_liquidity()
+
+                if should_stake:
+                    self.stake_liquidity(liquidity_tokens)
+
+                return liquidity_tokens
             else:
                 self.logger.warning(f"Failed to add liquidity with: {add_liquidity_args}")
         else:
             self.logger.info(f"Not enough tokens to add liquidity or liquidity already added")
 
-    def remove_liquidity(self) -> Optional[Receipt]:
+    def remove_liquidity(self, should_unstake: bool) -> Optional[Wad]:
```

Comment:
Yeah I was trying to support a range of scenarios in which, e.g. we may have already added liquidity and governance changes are made that support staking those liquidity tokens. Rather than having to submit multiple additional transactions, we could then simply just start adding liquidity with our existing LP tokens. That being said, these would more be edge cases that may not be hit so would also be open to attempting to simplify the logic.